helpers/premis.inc.php

```diff
@@ -42,32 +53,59 @@ function getUnderCopyrightId()
     return $id;
 }
 
-function addRights($objectId, $copyrightStatusId, $citationId, $actId, $restriction, $basisId = null)
+function addOrGetRights($objectId, $basisId = null)
 {
-    if (empty($basisId)) {
-        $basisId = QubitTerm::RIGHT_BASIS_POLICY_ID;
+  // Return existing rights if already linked to object
+  $criteria = new Criteria;
+  $criteria->add(QubitRelation::SUBJECT_ID, $objectId);
+  $criteria->add(QubitRelation::TYPE_ID, QubitTerm::RIGHT_ID);
```

Comment:
I think you can add a join here instead of doing two queries, e.g.

```
$criteria->addJoin(QubitRelation::OBJECT_ID, QubitRights::ID);
$rights = QubitRights::getOne($criteria);
```

Reply:
good call!